model/fileModel/file.go

```
@@ -0,0 +1,110 @@
+package fileModel
+
+import (
+	"fmt"
+	"context"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/bson"
+	// "go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+const (
+	dbName string = "test"
+	collectionName string = "files"
+)
+
+type File struct {
+	ID       primitive.ObjectID `bson:"_id, omitempty"`
+	Owner    string             `bson:"owner"`
+	Name     string             `bson:"name"`
+	Path     string             `bson:"path"`
+	IsFolder bool               `bson:"isFolder"`
+}
+
+func New(owner, name, path string, isFolder bool) File {
+	return File {
+		ID:    primitive.NewObjectID(),
+		Owner: owner,
+		Name:  name,
+		Path:  path,
+		IsFolder: isFolder,
+	};
+}
+
+func (f *File) Insert(client *mongo.Client) (*File, error) {
+    collection := client.Database(dbName).Collection(collectionName)
+    res, err := collection.InsertOne(context.TODO(), f)
+	if err != nil {
+        return nil, fmt.Errorf(
+			"Internal",
+			fmt.Sprintf("Internal error: %v", err),
+		)
+    }
+	f.ID = res.InsertedID.(primitive.ObjectID);
+	return f, nil;
+}
+
+func (f *File) Update(client *mongo.Client, filter bson.M) (*File, error) {
+	collection := client.Database(dbName).Collection(collectionName)
+	
+	updatedData := bson.M{};
+	fileBytes, _ := bson.Marshal(f)
```

You are ignoring the errors returned from `bson.Marshal` and `bson.Unmarshal`